client/src/App.js

```diff
@@ -14,16 +14,19 @@ export default class App extends Component {
 
   addToSavedList = movie => {
     const savedList = this.state.savedList;
-    savedList.push(movie);
-    this.setState({ savedList });
+    if (!savedList.some(x => x.id === movie.id)){
+      savedList.push(movie);
+      this.setState({ savedList }, () => console.log(this.state.savedList));
+    }
   };
 
   render() {
     return (
       <div>
         <SavedList list={this.state.savedList} />
-        <div>Replace this Div with your Routes</div>
-      </div>
+        <Route exact path="/" component={MovieList}/>
+        <Route path="/movies/:id" render={props => <Movie {...props} addToSavedList={this.addToSavedList} />} />
```

Comment:
Glad that you used Render Props, it will be a very useful thing in the future, so please try to dive deeper into this concept